Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -1,19 +1,30 @@
 import Constants from 'vtk.js/Sources/Widgets/Widgets3D/LineWidget/Constants';
 import macro from 'vtk.js/Sources/macro';
 import * as vtkMath from 'vtk.js/Sources/Common/Core/Math/';
-
 import {
   calculateTextPosition,
   updateTextPosition,
 } from 'vtk.js/Sources/Widgets/Widgets3D/LineWidget/helper';
 
+const {
+  Direction,
+  HandleBehavior,
+  HandleRepresentationType,
+  HandleVisibilityBehavior,
+} = Constants;
 const MAX_POINTS = 2;
 
-const { Direction, HandleBehavior, HandleRepresentationType } = Constants;
-
 export default function widgetBehavior(publicAPI, model) {
   model.classHierarchy.push('vtkLineWidgetProp');
-  let isDragging = null;
+
+  const moveHandle = model.widgetState.getMoveHandle();
+  const handle1 = model.widgetState.getHandle1();
+  const handle2 = model.widgetState.getHandle2();
+  const handle1Visibility = model.widgetState.getHandle1Visibility();
+  const handle2Visibility = model.widgetState.getHandle2Visibility();
+  const handle1FaceCamera = publicAPI.getWidgetState().getHandle1FaceCamera();
+  const handle2FaceCamera = publicAPI.getWidgetState().getHandle2FaceCamera();
+  const SVGLayerText = model.widgetState.getText();
```

Comment:
call `model.widgetState.getText()` each time you need it. Do not store it in this variable